resources/bolt_bash_completion.sh

```diff
@@ -6,134 +6,202 @@
 
-	case $prev in
-	*bolt)
-		next="command file task plan project group inventory secret script apply"
-		;;
-	command | script)
-		next="run"
-		;;
-	task)
-		next="show run"
-		;;
-	plan)
-		next="show run convert"
-		;;
-	file)
-		next="upload"
-		;;
-	group | inventory)
-		next="show"
-		;;
-	secret)
-		next="createkeys encrypt decrypt"
-		;;
-	encrypt | decrypt | createkeys)
-		next="-m --modulepath --plugin ${project_config_opts}"
-		[[ $prev == "createkeys" ]] && next="${next} --force"
-		;;
-	module)
-		next="install show generate-types"
-		;;
-	install)
-		next="--log-level -m --modulepath ${project_config_opts}"
-		;;
-	show-modules | generate-types)
-		next="--log-level -m --modulepath ${project_config_opts}"
-		;;
-	convert)
-		next="--log-level -m -modulepath ${project_config_opts}"
-		;;
-	show)
-		if [ "$prevprev" == "group" ]; then
-			next="${project_config_opts} -i --inventoryfile --log-level"
-		elif [ "$prevprev" == "inventory" ]; then
-			next="${targeting_opts} ${project_config_opts} --detail -i --inventoryfile --log-level"
-		elif [ "$prevprev" == "plan" ] || [ "$prevprev" == "task" ]; then
-			next="${general_opts} ${project_config_opts} -m --modulepath --filter --format"
-		fi
-		;;
-	run)
-		next="${all_options} ${targeting_opts} --params --tmpdir --ssh-command --copy-command"
-		[[ ${prevprev} == "plan" ]] && next="${next} --compile-concurrency --hiera-config"
-		;;
-	upload | apply)
-		next="${all_options} ${targeting_opts} --ssh-command --copy-command"
-		[[ ${prev} == "apply" ]] && next="${next} --compile-concurrency --hiera-config"
-		;;
-	project)
-		next="init migrate"
-		;;
-	init)
-		next="--log-level --modules"
-		;;
-	migrate)
-		next="--log-level -i --inventoryfile ${project_config_opts}"
-		;;
-	-t | --targets)
-		if [[ -f ${inventory_list_cache_file} ]]; then
-			local next=$(cat ${inventory_list_cache_file})
-		else
-			local jsondata=$(bolt --detail --targets all inventory show | sed -e '$d')
-			jsondata=${jsondata//\"/\\\"}
+          if [ -f "${PWD}/.plan_cache.json" ]; then
+            # Use Puppet's ruby instead of jq since we know it will be there.
+            next=$(get_json_keys "${PWD}/.plan_cache.json")
+          elif [ -f "${HOME}/.puppetlabs/bolt/.plan_cache.json" ]; then
+            next=$(get_json_keys "${HOME}/.puppetlabs/bolt/.plan_cache.json")
+          else
+            case $action in
+              convert) next="${context_opts}";;
+              new) next="--project --pp";;
+              run) next="${action_opts} ${apply_opts} --params --tmpdir";;
+              show) next="${context_opts} --filter --format";;
+            esac
+          fi
+        else
+          case $action in
+            convert) next="${context_opts}";;
+            new) next="--project --pp";;
+            run) next="${action_opts} ${apply_opts} --params --tmpdir";;
+            show) next="${context_opts} --filter --format";;
+          esac
+        fi
+        ;;
+      project)
```

Comment:
This needs a case for `plugin show` above now that https://github.com/puppetlabs/bolt/pull/2872 is merged.